pkg/apis/configuration/validation/virtualserver_test.go

```diff
@@ -1583,32 +1583,89 @@ func TestValidatePath(t *testing.T) {
 
 func TestValidateSplits(t *testing.T) {
```

Comment:
I would add a test to `TestValidateSplitsFails` as well to test that a negative number can't be used.

This might look like this:
```
{
	splits: []v1.Split{
		{
			Weight: 100,
			Action: &v1.Action{
				Pass: "test-1",
			},
		},
		{
			Weight: -2,
			Action: &v1.Action{
				Pass: "test-2",
			},
		},
	},
	upstreamNames: map[string]sets.Empty{
		"test-1": {},
		"test-2": {},
	},
	msg: "negative weight",
},
```

Reply:
test case added